pkg/virt-controller/watch/migration/migration.go

```diff
@@ -1406,59 +1555,76 @@ func (c *Controller) sync(key string, migration *virtv1.VirtualMachineInstanceMi
 			return nil
 		}
 
-		if !targetPodExists {
-			sourcePod, err := controller.CurrentVMIPod(vmi, c.podIndexer)
-			if err != nil {
-				log.Log.Reason(err).Error("Failed to fetch pods for namespace from cache.")
-				return err
-			}
-			if !controller.PodExists(sourcePod) {
-				// for instance sudden deletes can cause this. In this
-				// case we don't have to do anything in the creation flow anymore.
-				// Once the VMI is in a final state or deleted the migration
-				// will be marked as failed too.
-				return nil
-			}
-
-			if _, exists := migration.GetAnnotations()[virtv1.EvacuationMigrationAnnotation]; exists {
-				if err = descheduler.MarkEvictionInProgress(c.clientset, sourcePod); err != nil {
-					return err
+		if migration.IsLocalOrDecentralizedTarget() {
```

Comment:
```suggestion
		if !migration.IsLocalOrDecentralizedTarget() {
			break
		}
```
or `return nil`. Again to avoid big indent.

Reply:
I like return nil better, easier to understand what is going on.